### ISSUE TITLE
Missing head element causes stylesheet to not load after refresh

### ISSUE THREAD

**Issue by gvkhna** [withastro/astro#10151]
### Astro Info

```block
Astro                    v4.4.0
Node                     v20.10.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   server
Adapter                  @astrojs/node
Integrations             @astrojs/react
                         @astrojs/partytown
                         @astrojs/tailwind
                         @astrojs/mdx
                         astro-critters
                         astro-compress
                         astro-compressor
                         @astrojs/sitemap
                         astro-robots-txt
```

### Describe the Bug

Fun bug to track down. Was getting behavior where I would load the page and it looked fine. After one refresh the stylesheet (tailwind) would be gone. It simply was not included in subsequent requests. Turns out my Layout was the following:

```astro
---
import '@styles/tw.css'
---
<html>
<meta charset='UTF-8' />
    <meta
      name='viewport'
      content='width=device-width'
    />
<body>
<slot />
</body>
</html>
```

Only after a closer look I noticed the `<head></head>` was missing. Low priority but would be nice if astro caught that. I spent a while trying various things before noticing that.

### What's the expected result?

Error: Missing <head> element in Layout.astro can cause unexpected behavior.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-8zqmfz?file=src%2Flayouts%2FLayout.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by log101** [comment]
@gvkhna the example does not produce the error you mentioned. I also tried to produce the error myself with no success. I'd love to work on this bug if you could update the example.

**Comment by lilnasy** [comment]
With the layout you provided, styles and scripts are not rendered, but that behavior is the same in the first and the second request.

**Comment by gvkhna** [comment]
Yea it is not repro'ing the behavior I saw exactly and it's unclear exactly why it was so inconsistent for me. I would still gather a simple warning of styles or scripts are imported but head element is missing would be a great addition, or something even simpler, page body without head etc.

**Comment by matthewp** [comment]
@ascorbic this is a good one to update the repro and see if its still happening. If so we probably should talk about it.

**Comment by ascorbic** [comment]
I can't reproduce the issue. The example builds without a head, but the scripts and styles do load. For now we allow pages without a `<head>`, though we will likely deprecate support in a future version.
